ui/hooks/useTransactionInsights.js

```diff
@@ -29,8 +28,22 @@ const useTransactionInsights = ({ txData }) => {
   const { txParams, chainId, origin } = txData;
   const caip2ChainId = `eip155:${stripHexPrefix(chainId)}`;
   const insightSnaps = useSelector(getInsightSnaps);
-  const subjectMetadata = useSelector(getSubjectMetadataDeepEqual);
   const insightSnapIds = useSelector(getInsightSnapIds);
+  const state = useSelector((value) => value);
```

Comment:
I wonder if this will cause re-rendering problems, if so that is a merge blocker

Reply:
Potentially yeah. Do you have better ideas?

Reply:
Potentially we could create a selector that gets the manifests for all transaction insight snaps and use that instead of a function?

Reply:
Done.